experiment/simulation/js/animation-parallel-register.js

```diff
@@ -0,0 +1,807 @@
+}
+function appendI4() {
+    if (textI4.textContent != 0 && tl.progress() == 0) {
+        serialDisappear4();
+        textI4.textContent = 0;
+        svg.appendChild(textI4);
+        gsap.set(textI4, {
+            x: 46,
+            y: 654
+        });
+        gsap.set(I4, {
+
+            fill: "#eeeb22"
+        });
+        free();
+        serialVisible4();
+        errno();
+        setter(textI4.textContent, serialDot4);
+
+        OBSERV.innerHTML = "I4 is set to 0";
+    }
+    else if (textI4.textContent != 1 && tl.progress() == 0) {
+        appendI4To1();
+    }
+
+
+
+
+}
+function appendI4To1() {
+    serialDisappear4();
+    textI4.textContent = 1;
+    svg.appendChild(textI4);
+    gsap.set(textI4, {
+        x: 46,
+        y: 654
+    });
+    gsap.set(I4, {
+
+        fill: "#29e"
+    });
+    free();
+    serialVisible4();
+    errno();
+    setter(textI4.textContent, serialDot4);
+
+    OBSERV.innerHTML = "I4 is set to 1";
+
+}
+function appendClock() {
+    if (textClock.textContent != 0 && tl.progress() == 0) {
+        clockDisappear();
+        textClock.textContent = 0;
+        svg.appendChild(textClock);
+        gsap.set(textClock, {
+            x: 16,
+            y: 404
+        });
+        gsap.set(CLOCK, {
+
+            fill: "#eeeb22"
+        });
+        free();
+        clockVisible();
+        setter(textClock.textContent, clockDot);
+
+        errno();
+
+    }
+    else if (textClock.textContent != 1 && tl.progress() == 0) {
+        appendClockTo1();
+    }
+
+}
+function appendClockTo1() {
+    clockDisappear();
+    textClock.textContent = 1;
+    svg.appendChild(textClock);
+    gsap.set(textClock, {
+        x: 16,
+        y: 404
+    });
+    gsap.set(CLOCK, {
+
+        fill: "#29e"
+    });
+    free();
+    clockVisible();
+    setter(textClock.textContent, clockDot);
+
+    errno();
+    OBSERV.innerHTML = "Clock has Started";
+
+}
+function reboot() {
+    textI1.textContent = 2;
+    textI2.textContent = 2;
+    textI3.textContent = 2;
+    textI4.textContent = 2;
+
+    textClock.textContent = 2;
+
+}
+
+
+
+
+
+
+function outputSetter() {
+    setter(textO1.textContent, O1);
+    setter(textO2.textContent, O2);
+    setter(textO3.textContent, O3);
+    setter(textO4.textContent, O4);
+}
+
+function errno() {
+
+}
+function batado() {
+    OBSERV.innerHTML = "Simulation has finished. Press Restart to start again"
+}
+function setter(a, b) {
+    if (a == 1) {
+        unset(b);
+
+    }
+    else if (a == 0) {
+        set(b);
+    }
+}
+outputDisappear();
+var tl = gsap.timeline({ repeat: 0, repeatDelay: 0 });
+
+function fourXspeed() {
+    if (textClock.textContent == 1 && textI1.textContent != 2 && textI2.textContent != 2 && textI3.textContent != 2 && textI4.textContent != 2 && tl.progress() != 1) {
+        tl.resume();
+        tl.timeScale(4);
+        OBSERV.innerHTML = "4x speed";
+        decide = 1;
+        BUTTON.innerHTML = "Halt";
+    }
+}
+function doubleSpeed() {
```

Comment:
Do we need 2 separate functions with almost identical code for 2x and 4x speeds?